File: backend/Models/Attempt.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;


namespace prid_2324_a12.Models;

public class Attempt {
    
    [Key]
    public int Id {get; set;}
    public DateTimeOffset? Start { get; set;}
    public DateTimeOffset? Finish {get; set;}

    [ForeignKey(nameof(UserId))]
    public User User{get;set;} = null!;
    public int UserId {get;set;}

    [ForeignKey(nameof(QuizId))]   [JsonIgnore]
    public Quiz Quiz {get;set;} = null!;
    public int QuizId {get;set;}

    public virtual ICollection<Answer> Answers {get;set;}= new HashSet<Answer>();

}

File: frontend/src/app/models/Attempt.ts
import { Type } from "class-transformer";
import 'reflect-metadata';
import { Quiz } from "./quiz";
import { Solution } from "./solution";
import { Database } from "./database";
import { Answer } from "./Answer";


export class Attempt{
    id?: number;
    //quiz?: Quiz;
    Start?: Date;
    Finish?: Date;
    userId?:number;
    quizId?: number;
    answer?: Answer[];

    get display(): string {
        return `${this.id}`;
    }

    // get evaluation(): number{
        
    // }
}